server/api/batches.py

```diff
@@ -201,17 +203,20 @@ def validate_batch_results(
             f"Tally sheet names must be unique. '{duplicate_tally_sheet_name}' has already been used."
         )
 
-    total_votes = sum(
-        sum(tally_sheet["results"].values()) for tally_sheet in batch_results
-    )
-    assert contest.votes_allowed is not None
-    allowed_votes = batch.num_ballots * contest.votes_allowed
-    if total_votes > allowed_votes:
-        raise BadRequest(
-            f"Total votes for batch {batch.name} should not exceed {allowed_votes}"
-            f" - the number of ballots in the batch ({batch.num_ballots})"
-            f" times the number of votes allowed ({contest.votes_allowed})."
-        )
+    for contest in contests:
+        contest_choice_ids = set(choice.id for choice in contest.choices)
+        total_votes = 0
+        for tally_sheet in batch_results:
+            for choice_id, count in tally_sheet["results"].items():
+                total_votes += count if choice_id in contest_choice_ids else 0
```

Comment:
maybe simplify by iterating over `contest.choices` here

Reply:
Went ahead and made the switch, just noting that I sometimes find nested list comprehensions to be hard to grok: https://github.com/votingworks/arlo/pull/1884/commits/f7c59dc11649a596050f13a54ba2c06d6c7d1aad

Reply:
Good catch! https://github.com/votingworks/arlo/pull/1884/commits/16e7995f9b18847e32259c919d8565994d7b14bb